### PR TITLE
Apply recording-work relationship attributes to checked works

### LINKED ISSUE #32: mb-reledit-set_relation_attrs: Button have no effects
Same as [IvanDobsky](https://community.metabrainz.org/t/how-do-i-add-a-relationship-live-recording/524040/13?u=jesus2099) with Vivaldi 3.6.2165.40 (Chromium 88.0.4324.186) and Firefox 68.0 on Windows 10 (I was working late).

Clicking buttons does nothing, no errors.
As if event listeners were not there any more.

I got it randomly working but only 2% of the times.

~~I bet it's due to [React hydrate](https://github.com/jesus2099/konami-command/issues?q=label%3Areact-hydrate).~~

### PR DESCRIPTION
**Pull request by jesus2099**
Still apply them to checked recording if no checked works
Closes #32

### CODE REVIEW

**Comment by jesus2099** [comment]
I think it's more intuitive to apply recording-work changes to checked works.
But if no works are checked, it falls back to current behaviour of applying recording-work changes to checked recordings. :)

**Comment by jesus2099** [comment]
I added a cosmetic fix, as I was at it.

File: mb-reledit-set_relation_attrs.user.js [hunk 1]
```diff
@@ -41,7 +48,7 @@ function setAttributes(relationType, attrId, toggle) {
 (function displayToolbar() {
     relEditor.container(document.querySelector('div.tabs'))
              .insertAdjacentHTML('beforeend', `
-        <h3><span id="relattrs_script_toggle">▶ Relation attributes</span></h3>
+        <h3><span id="relattrs_script_toggle" style="cursor:pointer">▶ Relation attributes</span></h3>
```
Comment (jesus2099) [inline mb-reledit-set_relation_attrs.user.js]: @loujine, do you prefer the following, to pass this more than 100 characters check?

```html
        <h3>
            <span id="relattrs_script_toggle" style="cursor:pointer">▶ Relation attributes</span>
        </h3>
```

**Comment by jesus2099** [comment]
Or I thought about another alternate simple enhancement instead of this PR.
Maybe if no recordings are selected, you could pop up an alert saying something like _Please select at least one recording_. :)

File: mb-reledit-set_relation_attrs.user.js (hunk 1 above)
Comment (loujine) [reply mb-reledit-set_relation_attrs.user.js]: Yes, that looks good. I also usually use a 2-spaces indentation on HTML in JS files (just fixed it on that file with your other PR)

**Comment by loujine** [comment]
I never thought about using the script with the checked works but it seems logical, so I'm fine with merging this if it works.

Could you also add a new test? Just copying and changing https://github.com/loujine/musicbrainz-scripts/blob/master/tests/test_reledit.py#L112 to 'td.work input' will probably work

**Comment by jesus2099** [comment]
> Could you also add a new test? Just copying and changing https://github.com/loujine/musicbrainz-scripts/blob/master/tests/test_reledit.py#L112 to 'td.work input' will probably work

I understand what it aims to do, but I don't understand the code.
I'm sorry, I really don't know how to make the test and its check.

https://github.com/loujine/musicbrainz-scripts/blob/7b8056ffc4528b4b6f6087115606cac356bb5342/tests/test_reledit.py#L112-L116

Maybe I should duplicate these lines, replacing `recording` by `works` (yes, with an `s`):

```js
self.driver.find_element_by_css_selector('td.works input').click() 
self.driver.find_element_by_id('setLive').click() 
assert '>live recording of<' in self.driver.page_source 
self.driver.find_element_by_id('toggleLive').click() 
assert '>live recording of<' not in self.driver.page_source
```

🤔 ❓ 

But hmmm.
I often deal with code I don't really understand and replace some words with others.
But here, it seems I don't understand enough. 🤔

**Comment by loujine** [comment]
Yes, selenium is simulating a user navigating the website (it uses chrome without a screen) so
```
# click on all work checkboxes
self.driver.find_element_by_css_selector('td.works input').click() 
# click the "set live" button
self.driver.find_element_by_id('setLive').click() 
# make sure "<span ...">live recording of</span>" is now part of the source code
assert '>live recording of<' in self.driver.page_source 
```
etc.

I should probably have written the tests with javascript + selenium but it is faster to set things up with Python. 
Don't worry, I'll add a test if you don't manage to have one working

**Comment by jesus2099** [comment]
I tried adding that test. 😅 
You should squash all the commits in the end, I guess.

Maybe it would still be more simple to do [the alert thing](https://github.com/loujine/musicbrainz-scripts/pull/34#issuecomment-805346941). ;)

**Comment by jesus2099** [comment]
I _resolved conflicts_ within this PR page tool, directly.

**Comment by loujine** [comment]
Rebased, cleaned up a bit and completed the test. Not sure why the CI doesn't start in github actions, but the test runs fine on my machine

**Comment by loujine** [comment]
Thanks for the improvements!

**Comment by jesus2099** [comment]
You're welcome.
Wow, you could edit my branch! 😁

**Pull request merged**